### PR TITLE
build: add semantic release

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- adds semantic release pipeline steps to CI workflow, along lines of https://github.com/guardian/commercial-core
- updates CI steps to manually pull node version out of `.nvmrc`, because [`nvm install` doesn’t actually change the node version for following steps](https://github.com/guardian/commercial-core/runs/1258855238?check_suite_focus=true#step:5:6) in github actions

### CODE REVIEW

**Review by mxdvl — APPROVED** [review]
I like this.

File: docs/01-development-instructions.md [hunk 1]
```diff
@@ -60,12 +60,16 @@ VSCode should prompt you to install our recommended extensions when you open the
 
 You can also find these extensions by searching for `@recommended` in the extensions pane.
 
-## Publishing to NPM
+## Releasing
 
-The `consent-management-platform` is available to import from [NPM](https://www.npmjs.com/package/@guardian/consent-management-platform), You must be a member of the `guardian` organisation on NPM to publish new versions of this package.
+Changes are automatically released to NPM.
```
Comment (mxdvl) [inline docs/01-development-instructions.md:65]:
```suggestion
Changes are automatically released to [NPM][].

[NPM]: https://www.npmjs.com/package/@guardian/consent-management-platform
```

Comment (sndrs) [reply docs/01-development-instructions.md:65]: oooh i didn't know you could do this! 👍

**Comment by mxdvl** [comment]
Still some teething issues…

- 97b1ec06f149a3398af0d90282f77b6a37121253 did not trigger a release
- nor did 5c24f3e6e14e181ec6183b5363a78d396ca07699
- but b6b69d98d40b3cb67dbdf98753cd2a513457e5c0 did! 🍾

**Pull request merged**